processing/src/main/java/io/druid/query/extraction/LookupExtractionFn.java

```diff
@@ -47,7 +48,9 @@ public LookupExtractionFn(
       @JsonProperty("replaceMissingValueWith")
       final String replaceMissingValueWith,
       @JsonProperty("injective")
-      final boolean injective
+      final boolean injective,
+      @JsonProperty("optimize")
+      Boolean optimize
```

Comment:
i am assuming by default ppls would like thing without any optimization, i can switch it to on by default, i don't have a strong opinion about it.